### ISSUE TITLE
Explicit error conditions in `FieldExport`

### ISSUE THREAD

**Issue by RedbackThomson** [aws-controllers-k8s/community#1580]
**Is your feature request related to a problem?**
When reconciling the `FieldExport` custom resource, the controller does not propagate any errors up into the conditions. Therefore, if there is a typo in the `GroupKind`, or a missing destination `ConfigMap` resource, the only indication that something went wrong is in the controller logs.

**Describe the solution you'd like**
Any controller logs should be propagated to the resource conditions in the same way as other ACK resources so that users are able to detect and resolve errors without requiring access to pod logs.

**Describe alternatives you've considered**
N/A

**Comment by acornett21** [comment]
@RedbackThomson Could this also be the case for `AdoptedResource` too? Since it has its own reconciler, and is considered an `Additive/Optional` CR?

**Comment by RedbackThomson** [comment]
> Could this also be the case for `AdoptedResource` too?

Absolutely! I think we have some very minimal condition handling in `AdoptedResource`, but that definitely needs some more love and attention

**Comment by acornett21** [comment]
@RedbackThomson Should we duplicate this issue for `AdoptedResource` or use this one as a catchall?

**Comment by RedbackThomson** [comment]
> @RedbackThomson Should we duplicate this issue for `AdoptedResource` or use this one as a catchall?

I will duplicate so we can track them separately
